src/GraphQL/Builders/ConnectionBuilder_Typed.cs

```diff
@@ -0,0 +1,291 @@
+using System;
+#if NETSTANDARD2_1
+using System.Diagnostics.CodeAnalysis;
+#endif
+using System.Linq;
+using System.Threading.Tasks;
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+
+#nullable enable
+
+namespace GraphQL.Builders
+{
+    /// <summary>
+    /// Builds a connection field for graphs that have the specified source and return type.
+    /// </summary>
+    public class ConnectionBuilder<TSourceType, TReturnType>
+    {
+        private bool _isBidirectional => FieldType.Arguments.Any(x => x.Name == "before");
```

Comment:
```suggestion
        private bool IsBidirectional => FieldType.Arguments.Any(x => x.Name == "before");
```

It is a property, not field.

Reply:
It is private.  And renaming `_pageSize` to `PageSize` is not possible due to a naming conflict.

Reply:
It's private but it is stiill propperty. Property should follow PascalCase naming.